full-stack-tests/ecschema-rpc-interface/src/frontend/setup/IModelSession.ts

```diff
@@ -43,22 +44,32 @@ export class IModelSession {
       else if (iTwinList.length > 1)
         throw new Error(`Multiple iTwins named ${iModelData.iTwinName} were found for the user.`);
 
-      contextId = iTwinList[0].id;
+      iTwinId = iTwinList[0].id;
     } else
-      contextId = iModelData.iTwinId!;
+      iTwinId = iModelData.iTwinId!;
 
     if (iModelData.useName) {
-      const imodelClient = new IModelHubClient();
-      const imodels = await imodelClient.iModels.get(requestContext, contextId, new IModelQuery().byName(iModelData.name!));
-      if (undefined === imodels || imodels.length === 0)
-        throw new Error(`The iModel ${iModelData.name} does not exist in project ${contextId}.`);
-      imodelId = imodels[0].wsgId;
+      const imodelClient = new IModelsClient({ api: { baseUrl: `https://${process.env.IMJS_URL_PREFIX ?? ""}api.bentley.com/imodels`}});
+      const iModels = imodelClient.iModels.getRepresentationList({
+        authorization: AccessTokenAdapter.toAuthorizationCallback(await IModelApp.getAccessToken()),
+        urlParams: {
+          projectId: iTwinId,
+        },
```

Comment:
Could be filtered by [name](https://github.com/iTwin/imodels-clients/blob/main/clients/imodels-client-management/src/operations/imodel/IModelOperationParams.ts#L22), then we wouldn't need to iterate and search on the client side.

Reply:
thanks, just updated